third_party/move/tools/move-mutator/src/operator.rs

```diff
@@ -21,14 +23,10 @@ impl MutantInfo {
     }
 }
 
-/// The mutation operator to apply.
-#[derive(Debug, Copy, Clone)]
-pub enum MutationOperator {
-    BinaryOperator(BinOp),
-    UnaryOperator(UnaryOp),
-}
-
-impl MutationOperator {
+/// Trait for mutation operators.
+/// Mutation operators are used to apply mutations to the source code. To keep adding new mutation operators simple,
+/// we use a trait that all mutation operators implement.
+pub trait MutationOperatorTrait {
```

Comment:
```suggestion
pub trait MutationOperator {
```

But the
```rust
pub enum MutationOperator {
```
should be renamed to
```rust
pub enum Mutation {
// And the API would still make sense with this new name:
/// Mutation.apply(...)
/// Mutation.file_hash(...)
/// Mutation.op_type(...) ---> if you apply the below huge comment

```

Reply:
Renamed to `MutationOp`.